src/lib.rs

```diff
@@ -298,7 +298,7 @@ enum NodeKind<'input> {
     Element {
         tag_name: ExpandedNameOwned<'input>,
         attributes: Range,
-        namespaces: Range,
+        namespaces: Vec<usize>,
```

Comment:
I'm not sure that this is a good idea. I prefer one large allocation to multiple tiny one. Afaik, the benchmarks can be affected drastically by the used allocator. So it's hard to say that this is actually better before testing on different OS'es with different allocators.

Reply:
Hmmmm... good point that lots of tiny allocations take place here. Is there the potential to limit the number of namespace in scope? I'll keep thinking :thinking:

Reply:
What kind of limit? The maximim number of namespaces per element? I don't that any kind of limit is a good idea. Plus, even 3-5 namespaces in an XML with thousands of elements will blow up the memory.

As always, we have to choose between memory and CPU usage. Right now, `roxmltree` leans to a lower CPU usage, which is basically zero in the current implementation (at least when we work with namespaces).